src/SAML2/XML/saml/SubjectConfirmation.php

```diff
@@ -138,7 +138,7 @@ public static function fromXML(DOMElement $xml): object
         Assert::same($xml->namespaceURI, SubjectConfirmation::NS);
 
         if (!$xml->hasAttribute('Method')) {
-            throw new \Exception('SubjectConfirmation element without Method attribute.');
+            throw new \InvalidArgumentException('SubjectConfirmation element without Method attribute.');
```

Comment:
This one can also be expressed with webmozart, if we want (I've done that in some places):

```php
        Assert::true($xml->hasAttribute('Method'), 'SubjectConfirmation element without Method attribute.');
```